src/client.cpp

```diff
@@ -59,7 +60,8 @@ extern "C" {
 
 void ADDON_ReadSettings(void)
 {
-  Settings::GetInstance().ReadSettings();
+  settings = new tvheadend::Settings();
```

Comment:
Despite from having another settings singleton now (which makes no sense if your goal is to eleiminate it), this introduces a memory leak in case ADDON_ReadSettings gets called more than once. And even it is called only only once, where gets the global `settings` instance deleted?